### PR TITLE
Can 922

### PR DESCRIPTION
**Pull request by array-junkie**
Api for camp history

### CODE REVIEW

File: app/Models/Support.php [hunk 1]
```diff
@@ -373,4 +373,16 @@ public static function getSupportedCampsList($topicNum, $user_id)
         $result = DB::select($query);
         return $result;
     }
+
+    public static function ifIamImplicitSupporter($filter,$nickNames,$submit_time = null){
+        $liveCamp = Camp::getLiveCamp($filter);
+        $allChildCamps = Camp::getAllChildCamps($liveCamp);
+        if($submit_time){
+            $support = self::where('topic_num','=',$filter['topicNum'])->whereIn('camp_num',$allChildCamps)->whereIn('nick_name_id',$nickNames)->where('delegate_nick_name_id',0)->where('end','=',0)->where('start','<=',$submit_time)->first();
```
Comment (akuks) [inline app/Models/Support.php:381]: How about this
```php
$support = self::where('topic_num','=',$filter['topicNum'])->whereIn('camp_num',$allChildCamps)->whereIn('nick_name_id',$nickNames)->where('delegate_nick_name_id',0)->where('end','=',0)
->where(function($query) use ($submit_time)
{
    if ($submit_time) {
        $query->where('start','<=',$submit_time);
    }
})->first()

```

Comment (akuks) [reply app/Models/Support.php:381]: Can you please test it

**Review by akuks — CHANGES_REQUESTED** [review]

File: app/Models/Support.php (hunk 1 above)
Comment (M-Darab) [reply app/Models/Support.php]: Yes its working perfect. Thanks

**Review by akuks — APPROVED** [review]

**Comment by akuks** [comment]
@M-Darab Can you plese resolve the conflicts?

**Pull request merged**